src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -301,7 +303,20 @@ private static TypeSymbol NarrowObjectType(ITypeManager typeManager, ObjectSynta
 
             if (missingRequiredProperties.Any())
             {
-                diagnostics.Add(DiagnosticBuilder.ForPosition(expression).MissingRequiredProperties(ShouldWarn(targetType), missingRequiredProperties));
+                TextSpan span = expression.Span;

```

Comment:
You can avoid requesting span with `IPositionable` e.g.:
```
IPositionable positionable = expression;
```
And later on:
```
positionable = p1.Key; 
```
etc.

Reply:
done